lib/settings.schema.json

```diff
@@ -0,0 +1 @@
(can be left empty)","default":""},"variablePrefix":{"type":"string","description":"Prefix for variables (can be left empty)","default":""},"variableSuffix":{"type":"string","description":"Suffix for variables (can be left empty)","default":""},"objectCasing":{"type":"string","description":"Casing for all non-constant objects (signals, variables, entities etc.).","default":"ignore","enum":["snake_case","PascalCase","camelCase","CONSTANT_CASE","ignore"]},"constantGenericCasing":{"type":"string","description":"Casing for constants and generics.","default":"ignore","enum":["snake_case","PascalCase","camelCase","CONSTANT_CASE","ignore"]},"constantPrefix":{"type":"string","description":"Prefix for constants (can be left empty)","default":""},"constantSuffix":{"type":"string","description":"Suffix for constants (can be left empty)","default":""},"genericPrefix":{"type":"string","description":"Prefix for generics (can be left empty)","default":""},"genericSuffix":{"type":"string","description":"Suffix for generics (can be left empty)","default":""},"outPrefix":{"type":"string","description":"Prefix for port outputs (can be left empty).","default":""},"outSuffix":{"type":"string","description":"Suffix for port outputs (can be left empty).","default":""},"inPrefix":{"type":"string","description":"Prefix for port inputs (can be left empty).","default":""},"inSuffix":{"type":"string","description":"Suffix for port inputs (can be left empty).","default":""},"inoutPrefix":{"type":"string","description":"Prefix for port inouts (can be left empty).","default":""},"inoutSuffix":{"type":"string","description":"Suffix for port inouts (can be left empty).","default":""},"parameterOutPrefix":{"type":"string","description":"Prefix for parameter outputs (can be left empty).","default":""},"parameterOutSuffix":{"type":"string","description":"Suffix for parameter outputs (can be left empty).","default":""},"parameterInPrefix":{"type":"string","description":"Prefix for parameter inputs (can be left empty).","default":""},"parameterInSuffix":{"type":"string","description":"Suffix for parameter inputs (can be left empty).","default":""},"parameterInoutPrefix":{"type":"string","description":"A Prefix for parameter inouts (can be left empty).","default":""},"parameterInoutSuffix":{"type":"string","description":"Suffix for parameter inouts (can be left empty).","default":""},"labelCasing":{"type":"string","description":"Casing for labels.","default":"lowercase","enum":["lowercase","UPPERCASE","ignore"]},"instantiationLabelPrefix":{"type":"string","description":"Prefix for instantiation labels (can be left empty)","default":""},"instantiationLabelSuffix":{"type":"string","description":"Suffix for instantiation labels (can be left empty)","default":""},"portOmission":{"type":"boolean","description":"Enforce port omission rule: Ports may not be implicitly left unconnected. Explicit open must be used.","default":false}},"additionalProperties":false}},"additionalProperties":false}
```

Comment:
As said before this should not be part of the repository and be auto generated during build. (and maybe for debugging but be gitignored than)